### PR TITLE
Nicer ManifestInfo check

### PR DESCRIPTION
**Pull request by ignasi35**
References #30044

### CODE REVIEW

File: akka-actor-tests/src/test/scala/akka/util/ManifestInfoSpec.scala [hunk 1]
```diff
@@ -18,6 +18,19 @@ class ManifestInfoSpec extends AkkaSpec {
...
+    "support dynver" in {
+      val versions = Map(
+        "akka-actor" -> new ManifestInfo.Version("2.6.4"),
+        "akka-persistence" -> new ManifestInfo.Version("2.6.4+10-abababef"))
+      val allModules = List("akka-actor", "akka-persistence")
+      ManifestInfo.checkSameVersion("Akka", allModules, versions) shouldBe Some(
+        "You are using version 2.6.4+10-abababef of Akka, but it appears you (perhaps indirectly) also depend on older versions of related artifacts. " +
+        "You can solve this by adding an explicit dependency on version 2.6.4+10-abababef of the [akka-actor] artifacts to your project. " +
+        "Here's a complete collection of detected artifacts: (2.6.4, [akka-actor]), (2.6.4+10-abababef, [akka-persistence]). " +
         "See also: https://doc.akka.io/docs/akka/current/common/binary-compatibility-rules.html#mixed-versioning-is-not-allowed")
```
Comment (ignasi35) [inline akka-actor-tests/src/test/scala/akka/util/ManifestInfoSpec.scala:34]: I had some unexpected output on a test run in a separate project and, for a moment, suspected dynver was not supported.

I've left the test I added but I don't think it's a necessary test and would happily remove it.

File: akka-actor/src/main/scala/akka/util/ManifestInfo.scala [hunk 2]
```diff
@@ -80,11 +80,18 @@ object ManifestInfo extends ExtensionId[ManifestInfo] with ExtensionIdProvider {
...
+        .map { case (k, v) => k -> v.map(_._1).mkString("[", ", ", "]") }
+        .map { case (k, v) => s"($k, $v)" }
+        .mkString(", ")
       Some(
         s"You are using version $highestVersion of $productName, but it appears " +
         s"you (perhaps indirectly) also depend on older versions of related artifacts. " +
         s"You can solve this by adding an explicit dependency on version $highestVersion " +
         s"of the [$toBeUpdated] artifacts to your project. " +
+        s"Here's a complete collection of detected artifacts: ${groupedByVersion}. " +
         "See also: https://doc.akka.io/docs/akka/current/common/binary-compatibility-rules.html#mixed-versioning-is-not-allowed")
```
Comment (ignasi35) [inline akka-actor/src/main/scala/akka/util/ManifestInfo.scala:95]: This informs of all artifacts considered in the decision process but doesn't improve the message re what alternatives the user has and what is the preferred alternative in most cases (keep the highest akka version).

**Review by ignasi35 — COMMENTED** [review]
I'm keeping this review DRAFT in case we think it needs some extra attention on the docs or elsewhere.

File: akka-actor/src/main/scala/akka/util/ManifestInfo.scala [hunk 3]
```diff
@@ -80,11 +80,18 @@ object ManifestInfo extends ExtensionId[ManifestInfo] with ExtensionIdProvider {
     if (values.size > 1) {
       val highestVersion = values.max
       val toBeUpdated = filteredVersions.collect { case (k, v) if v != highestVersion => s"$k" }.mkString(", ")
+      val groupedByVersion = filteredVersions.toSeq
+        .groupBy { case (_, v) => v.version }
+        .view
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/ManifestInfo.scala:85]: iterator instead of view ?

Comment (patriknw) [reply akka-actor/src/main/scala/akka/util/ManifestInfo.scala:85]: should we also sort it? sort by version, and also sort the list of names

**Review by patriknw — COMMENTED** [review]
looking good

File: akka-actor-tests/src/test/scala/akka/util/ManifestInfoSpec.scala (hunk 1 above)
Comment (ignasi35) [reply akka-actor-tests/src/test/scala/akka/util/ManifestInfoSpec.scala:34]: Heh! turns out the dynver test caught an edge case as I was introducing support to sort the output.

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]
LGTM
